src/Forms/FormMessage.php

```diff
@@ -92,6 +91,30 @@ public function setMessage(
         return $this;
     }
 
+    /**
+     * Appends a message to the existing message if the types match, overrides if not
+     */
+    public function appendMessage(
+        string $message,
+        string $messageType = ValidationResult::TYPE_ERROR,
+        string $messageCast = ValidationResult::CAST_TEXT,
+    ): static {
+        if (!empty($this->message)) {
+            $canBeMerged = ($messageType === $this->getMessageType() && $messageCast === $this->getMessageCast());
+            // Checks that the exact message string is not already contained before appending
+            if ($canBeMerged && strpos($this->message, $message) === false) {
+                $message = sprintf(
+                    '%s%s%s',
+                    $this->message,
+                    $messageType === ValidationResult::CAST_HTML ? '<br />' : PHP_EOL,
```

Comment:
```suggestion
                    $messageCast === ValidationResult::CAST_HTML ? '<br />' : PHP_EOL,
```

Also, what happens if someone changes the cast in some `FormField` or injected `Form` by changing the value of `$messageCast` after the message is appended?

Reply:
Good question - that's currently a possibility, right? You could set an HTML message and then change the cast, is that a problem we want to solve here?

I can see adding some `nl2br()`-style logic into `setMessageCast()` but that feels outside the scope

Reply:
> You could set an HTML message and then change the cast, is that a problem we want to solve here?

It's the (potential) problem I'm pointing out, but I'm not sure if we need to solve it, or if we should just be aware of it and accept that people doing weird things will have weird results.

Probably the fact that we don't have a setter method for it is enough to say "we don't support changing that property directly", and in that case what you have here is good.